test/new-e2e/system-probe/test/system-probe-test_spec.go

```diff
@@ -0,0 +1,261 @@
+}
+
+type testConfig struct {
+	bundle         string
+	env            map[string]interface{}
+	filterPackages filterPaths
+}
+
+type filterPaths struct {
+	paths     []string
+	inclusive bool
+}
+
+var skipPrebuiltTests = filterPaths{
+	paths:     []string{"pkg/collector/corechecks/ebpf/probe"},
+	inclusive: false,
+}
+
+var runtimeCompiledTests = filterPaths{
+	paths: []string{
+		"pkg/network/tracer",
+		"pkg/network/protocols/http",
+		"pkg/collector/corechecks/ebpf/probe",
+	},
+	inclusive: true,
+}
+
+var coreTests = filterPaths{
+	paths: []string{
+		"pkg/collector/corechecks/ebpf/probe",
+		"pkg/network/protocols/http",
+	},
+	inclusive: true,
+}
+
+var fentryTests = filterPaths{
+	paths:     skipPrebuiltTests.paths,
+	inclusive: false,
+}
+
+func pathEmbedded(fullPath, embedded string) bool {
+	normalized := fmt.Sprintf("/%s/",
+		strings.TrimRight(
+			strings.TrimLeft(embedded, "/"),
+			"/",
+		),
+	)
+
+	return strings.Contains(fullPath, normalized)
+}
+
+func glob(dir, filePattern string, filter filterPaths) ([]string, error) {
+	var matches []string
+	err := filepath.WalkDir(dir, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		present, err := regexp.Match(filePattern, []byte(d.Name()))
+		if err != nil {
+			return err
+		}
+
+		if d.IsDir() || !present {
+			return nil
+		}
+		for _, p := range filter.paths {
+			if pathEmbedded(path, p) && filter.inclusive {
+				matches = append(matches, path)
+			} else if !pathEmbedded(path, p) && !filter.inclusive {
+				matches = append(matches, path)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return []string{}, err
+	}
+
+	return matches, nil
+}
+
+func generatePacakgeName(file string) string {
+	pkg := strings.TrimLeft(
+		strings.TrimRight(
+			strings.TrimPrefix(
+				strings.TrimSuffix(file, Testsuite),
+				TestDirRoot,
+			),
+			"/"),
+		"/")
+
+	return pkg
+}
+
+func buildCommandArgs(file, bundle string) []string {
+	pkg := generatePacakgeName(file)
+	junitfilePrefix := strings.ReplaceAll(pkg, "/", "-")
+	xmlpath := filepath.Join(
+		"/", "tmp", bundle,
+		fmt.Sprintf("%s.xml", junitfilePrefix),
+	)
+	jsonpath := filepath.Join(
+		"/", "tmp", bundle,
+		fmt.Sprintf("%s.json", junitfilePrefix),
+	)
+	args := []string{
+		"--format", "dots",
+		"--junitfile", xmlpath,
+		"--jsonfile", jsonpath,
+		"--raw-command", "--",
+		"/go/bin/test2json", "-t", "-p", pkg, file, "-test.v", "-test.count=1",
+	}
+
+	return args
+}
+
+func mergeEnv(env ...map[string]interface{}) []string {
+	var mergedEnv []string
+
+	for _, e := range env {
+		for key, element := range e {
+			mergedEnv = append(mergedEnv, fmt.Sprintf("%s=%s", key, fmt.Sprint(element)))
+		}
+	}
+
+	return mergedEnv
+}
+
+func runCommandAndStreamOutput(cmd *exec.Cmd, commandOutput io.Reader) error {
+	go func() {
+		scanner := bufio.NewScanner(commandOutput)
+		for scanner.Scan() {
+			_, _ = os.Stdout.Write([]byte(scanner.Text() + "\n"))
```

Comment:
It seems like you are just passing the text straight to `stdout`, so why is this necessary?

Reply:
I have done this so we can stream the output rather than waiting for the command to complete before the output is printed.

Reply:
There is an easier way, you can do `cmd.Stdout = os.Stdout` and `cmd.Stderr = os.Stderr`